### PR TITLE
Feature branch powerbanks

### PR DESCRIPTION
**Pull request by TheVagrantKnight**
#### Summary
Features "Powerbanks now exist"

-----------------------------------------------
#### Purpose of change

The purpose is to add 2 new powerbank items to the game. Powerbanks are relatively common and so it was weird that we have no way of charging phones or other UPS items out in the field without a military grade UPS. These items have been added with a ratio of 1 powerbank / 3-4 phones in the loot tables. The powerbanks themselves should spawn at a ratio of roughly 3 normal powerbanks / 20 cheap powerbanks>

-----------------------------------------------
##### It adds 2 new items:
- a cheap powerbank which is the aquivalent of a 20000 mAh 
- a normal powerbank which is the aquivalent of a 36000 mAh 

##### how it works:
These powerbanks have a charge like normal batteries as opposed to working with a magazine well that can be filled with batteries, as does the UPS. This means these charges cannot be refilled unless the powerbank is hooked up to a functioning power network or placed in a battery charger.

When activating the battery, the player is given 2 options:
- turn on / off : if turned on, the powerbank will start charging other UPS devices. This is equivalent to hooking up a cable between powerbank and a phone irl. The name is left as is to avoid confusion
- plug in / manage cables: similar to other powered tools and is used to charge the powerbank if no battery charger is available.

##### spawns:
I added powerbanks to most locations where a smartphone could be found with a ratio of roughly 1 to 3. This includes locations as well as zombie drops.

##### technicalities:
the weight, volume and max length are based on the 2 powerbanks I own.
The charge capacity is based on a quick guide I found wherein was stated that 20000 mAh powerbanks could charge a phone 4 times, while 36000 mAh powerbanks could charge a phone 6 times. As such, the charges are 4 * 120 and 6 * 120 with 120 being the charge of the smartphone in game.

#### note:
as of yet, the powerbank will not light up yellow in the inventory when turned on. This is an issue with the IS_UPS tag that is need to charge other UPS devices and is also present in the current UPS item. 3/4 of the time spent on this was figuring this out and to fix the Issue I'd have to do a proper deep dive into how the IS_UPS tag affects the active property.

#### Describe alternatives you've considered

-----------------------------------------------

#### Testing

It should not cause any regression since this is a pure addition of new items. No existing code was touched and in the tests I performed I did not run into any issues.

Testing was done as follows:
1. start a new world with a new survivor
2. use debug to spawn in the cheap and normal powerbanks
afbeelding

3.  Hook them up to an ad-hoc power network to see if they charge 
afbeelding

4.Put them in a battery box to see if they charge 
afbeelding

afbeelding

5.debug in a phone and reduce charge to 0
afbeelding

6. try turning on both powerbanks one by one to see if they charge the phone
afbeelding

afbeelding

7. Now that item functionality was successfully tested, I teleported to the nearest city and debug killed a lot of zombies. The cheap powerbank was found a couple corpses. Both powerbanks are included in a distribution group and only this group is ever used to add powerbanks to loot tables. So I considered this test a success.

### CODE REVIEW

**Comment by Brambor** [comment]
Why `cheap powerbank` and `powerbank`? Cheap invokes some knock-off. Why not small/medium or medium/large etc.?

File: data/json/items/tool/electronics.json [hunk 1]
```diff
@@ -914,5 +914,109 @@
...
     }
+  },
+  {
+    "id": "20000mah_powerbank_off",
+    "type": "TOOL",
+    "name": { "str": "cheap powerbank (UPS) (off)", "str_pl": "cheap powerbanks (UPS) (off)" },
+    "description": "A cheap rechargable powerbank that was quite common before the cataclysm. Who would have thought it'd take an apocalypse to reduce screen time? You could use it to charge your other UPS devices.",
+    "weight": "412 g",
+    "volume": "220 ml",
+	"longest_side": "13 cm",
```
Comment (Brambor) [inline data/json/items/tool/electronics.json:925]: You marked all these bot messages as resolved. I assume you astyled locally and haven't pushed yet?

**Comment by TheVagrantKnight** [comment]
I used "cheap" since the two aren't that different when it comes to form factor. It's mostly just pricing and charge capacity. I wasn't too fond of the wording myself and wanted to check up on some things before continuing.

**Comment by emixa-d** [comment]
Assuming that these powerbanks are USB-A or USB-C (fairly standard for powerbanks), this would imply that laser rifles can get their energy from an USB connection (neat). 
That would limit the frequency of how often the rifle can shoot a lot, since the powerbank is only 20W. So, I'm interpreting this as the rifle having multiple connections, with the military UPS connection (whatever it is) being standard, and the USB connection as a backup when stranded in an urban environment.

However, this appears to be an unintended consequence of this PR, which is why I'm mentioning it, so you can make it intentional (or change the PR to not make this a consequence).

Also ignore what I wrote below, I didn't notice the "20 W" at first.

>A laser rifle uses 40 charges (= 40 kJ) per shot. Assuming a frequency of 0.4 shots/second (this is the max frequency without overheating), this is 16 kW -- so, that's 16 kW going over the USB cable.
>
>Maybe cata-Earth has better technology, but that's not going to work on civilian powerbanks.
>
>I think you will need to remove IS_UPS from the powerbanks. The civilian UPS item "shooter's ear plugs" can be 'recharged' by changing the battery. The smart watch, smartphone and laptop can perhaps be recharged by plugging-in the device into the powerbank (although I don't know if that is actually implemented, I haven't tried connecting items I'm carrying to each other).~~

**Comment by TheVagrantKnight** [comment]
The thing about the laser rifle is a fair point. A powerbank isn't exactly meant to sustain that. I'll see if I can work out the flag system and maybe add 2 new flags IS_CIVIL_CHARGER and USE_CIVIL_CHARGER (first thing that came to mind) which will have a slower recharge rate and won't be added to laser weaponry

**Comment by Brambor** [comment]
Why close? Did you mean to put it in draft to not be accidentally merged?

**Comment by emixa-d** [comment]
>The civilian UPS item "shooter's ear plugs" can be 'recharged' by changing the battery.

Correction, these don't have swappable battery.

**Pull request closed without merging**